### DISCUSSION TITLE
Improve operator restore resilience to cloud storage network bandwidth limits

### DISCUSSION THREAD

**Discussion by OliverHfm** [cloudnative-pg/cloudnative-pg#1326] [Ideas]
Upon cluster initialisation from backup we experience failures like the following: 

`ERROR: Barman cloud restore exception: An error occurred (SlowDown) when calling the GetObject operation (reached max retries: 4): Please reduce your request rate.`

The error is caused due to QoS (bandwidth) limits applied to our cloud storage (S3) by our storage administration team. Therefore it is clear as to why the restore is running into the error.

Unfortunately the restore job is failing after the above situation occurred because an empty PGDATA directory exists on the system. The operator does not perform a retry of the restore but enters a crash loop.

Hence we would like to suggest improving operator resilience for these kind of scenarios. In the above scenario we think the operator should retry the restore operation even though the PGDATA directory already exists.

**Comment by ba-edb** [comment]
@gbartolini is this a CNPG issue or a Barman issue? /cc @mikewallace1979

**Comment by mikewallace1979** [comment]
It's a problem that could be solved at either the Barman level or the CNPG level - there is some discussion over in #1327.

Regardless of whether CNPG implements bandwidth limits, I can see value in Barman using any bandwidth throttling capabilities provided by the cloud provider libraries so I have filed EnterpriseDB/barman#746.

**Comment by oliverhofmannedb** [comment]
@gbartolini - apart from the barman efforts in EnterpriseDB/barman#746, has the CNPG team concluded if adding a retry logic to cover this kind of scenario is a possible enhancement?
Or is this (empty PGDATA) considered a fatal scenario where a DBA is considered to take manual action instead of the operator attempting the restore again.
